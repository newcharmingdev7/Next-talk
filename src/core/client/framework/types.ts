// TODO: (@cvle) Extract useful common types into its own package.
export { Omit, Overwrite, PropTypesOf } from "coral-ui/types";
export { DeepPartial } from "coral-common/types";

export type RelayEnumLiteral<T> = keyof T | "%future added value";
