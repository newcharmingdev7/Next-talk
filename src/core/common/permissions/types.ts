export type UserRole =
  | "%future added value"
  | "COMMENTER"
  | "MEMBER"
  | "STAFF"
  | "MODERATOR"
  | "ADMIN";

export interface User {
  id: string;
  role: UserRole;
  moderationScopes?: null | {
    scoped?: boolean;
    siteIDs?: string[];
  };
}

export type Moderator = User & {
  role: "MODERATOR";
};

export type LTEModerator = User & {
  role: "MODERATOR" | "STAFF" | "MEMBER" | "COMMENTER";
};

export const isModerator = (user: User): user is Moderator => {
  return user.role === "MODERATOR";
};

export const isLTEModerator = (user: User): user is LTEModerator => {
  return user.role !== "ADMIN";
};
