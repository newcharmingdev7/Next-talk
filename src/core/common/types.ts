export type Diff<T extends keyof any, U extends keyof any> = ({ [P in T]: P } &
  { [P in U]: never } & { [x: string]: never })[T];

export type Omit<U, K extends keyof U> = Pick<U, Diff<keyof U, K>>;

export type Overwrite<T, U> = Pick<T, Diff<keyof T, keyof U>> & U;

export type Sub<T, U> = Pick<T, Diff<keyof T, keyof U>>;

/**
 * Make all properties in T writeable
 */
export type Writeable<T> = { -readonly [P in keyof T]: T[P] };

export type Promiseable<T> = Promise<T> | T;
