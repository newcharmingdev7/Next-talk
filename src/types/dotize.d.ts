declare module "dotize" {
  export function convert(obj: Object): Record<string, any>
}
