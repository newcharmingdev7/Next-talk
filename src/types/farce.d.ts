declare module "farce" {
  export class BrowserProtocol {}
  export class queryMiddleware {}
}
