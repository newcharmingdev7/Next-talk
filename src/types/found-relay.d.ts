declare module "found-relay" {
  import { Environment } from "relay-runtime";
  export class Resolver {
    constructor(relayEnvironment: Environment);
  }
}
