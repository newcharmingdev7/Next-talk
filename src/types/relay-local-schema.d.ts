declare module "relay-local-schema" {
  import { GraphQLSchema } from "graphql";
  import { Network as RelayNetwork, FetchFunction } from "relay-runtime";

  interface CreateArguments {
    schema: GraphQLSchema;
    rootValue?: any;
    contextValue?: any;
  }

  export class Network {
    static create(data: CreateArguments): RelayNetwork;
  }

  export function createFetch(data: CreateArguments): FetchFunction;
}
