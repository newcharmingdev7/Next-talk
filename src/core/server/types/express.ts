import { Request } from "express";

import { Tenant } from "talk-server/models/tenant";
import { User } from "talk-server/models/user";
import TenantCache from "talk-server/services/tenant/cache";

export interface Request extends Request {
  user?: User;
  tenant?: Tenant;
  tenantCache: TenantCache;
}
