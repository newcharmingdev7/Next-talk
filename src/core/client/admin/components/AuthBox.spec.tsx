import React from "react";
import { createRenderer } from "react-test-renderer/shallow";

import { PropTypesOf } from "talk-framework/types";

import AuthBox from "./AuthBox";

it("renders correctly", () => {
  const props: PropTypesOf<typeof AuthBox> = {
    title: "title",
    children: "content",
  };
  const renderer = createRenderer();

  renderer.render(<AuthBox {...props} />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
