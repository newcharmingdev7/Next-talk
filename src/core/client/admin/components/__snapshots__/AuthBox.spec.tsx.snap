// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  data-testid="authBox"
>
  <withPropsOnChange(Flex)
    justifyContent="center"
  >
    <withPropsOnChange(HorizontalGutter)
      className="AuthBox-container"
      size="double"
    >
      <withPropsOnChange(Flex)
        justifyContent="center"
      >
        <div
          className="AuthBox-brandIcon"
        >
          <withPropsOnChange(BrandIcon)
            size="lg"
          />
        </div>
      </withPropsOnChange(Flex)>
      <div>
        <withPropsOnChange(Typography)
          align="center"
          variant="heading3"
        >
          title
        </withPropsOnChange(Typography)>
        <withPropsOnChange(BrandName)
          align="center"
          size="lg"
        />
      </div>
      content
    </withPropsOnChange(HorizontalGutter)>
  </withPropsOnChange(Flex)>
</div>
`;
